feat: add `math/base/special/sinf`

anandkaranubc opened the pull request:
Progresses https://github.com/stdlib-js/stdlib/issues/649

## Description

> What is the purpose of this pull request?

This pull request:

-  Adds the complete implementation for `math/base/special/sinf`

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   Progresses https://github.com/stdlib-js/stdlib/issues/649

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

anandkaranubc committed:
feat: temp commit

anandkaranubc committed:
Merge branch 'develop' into feat/sinf

anandkaranubc committed:
feat: add `math/base/special/sinf`

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: passed
  - task: lint_repl_help
    status: passed
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: passed
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: passed
  - task: lint_c_examples
    status: passed
  - task: lint_c_benchmarks
    status: passed
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: passed
  - task: lint_typescript_tests
    status: passed
  - task: lint_license_headers
    status: passed
---

anandkaranubc commented on lib/node_modules/@stdlib/math/base/special/sinf/package.json:
```diff
@@ -0,0 +1,147 @@
+    "radians",
+    "angle"
+  ],
+  "__stdlib__": {
```
Added metadata too.

anandkaranubc commented on lib/node_modules/@stdlib/math/base/special/sinf/src/main.c:
```diff
@@ -0,0 +1,154 @@
+/**
+* Computes the sine of a single-precision floating-point number (in radians).
+*
+* ## Method
```
I kept this comment because the same approach is used here too, like `sin`

anandkaranubc commented on lib/node_modules/@stdlib/math/base/special/sinf/test/test.js:
```diff
@@ -0,0 +1,187 @@
+		x[ i ] = f32( x[ i ] );
+		expected[ i ] = f32( expected[ i ] );
+		y = sinf( x[ i ] );
+		t.strictEqual( y, expected[ i ], 'returns expected value' );
```
No need of approximate tests.

anandkaranubc committed:
Update main.c

Signed-off-by: Karan Anand <119553199+anandkaranubc@users.noreply.github.com>

anandkaranubc committed:
Update main.c

Signed-off-by: Karan Anand <119553199+anandkaranubc@users.noreply.github.com>

Planeshifter commented on lib/node_modules/@stdlib/math/base/special/sinf/manifest.json:
```diff
@@ -0,0 +1,96 @@
+      "dependencies": [
+        "@stdlib/math/base/napi/unary",
+        "@stdlib/number/float32/base/to-word",
+        "@stdlib/constants/float64/high-word-abs-mask",
```
Appears to be unused.

```suggestion
```

Planeshifter commented on lib/node_modules/@stdlib/math/base/special/sinf/manifest.json:
```diff
@@ -0,0 +1,96 @@
+      "libpath": [],
+      "dependencies": [
+        "@stdlib/number/float32/base/to-word",
+        "@stdlib/constants/float64/high-word-abs-mask",
```
Appears to be unused.

```suggestion
```

Planeshifter commented on lib/node_modules/@stdlib/math/base/special/sinf/manifest.json:
```diff
@@ -0,0 +1,96 @@
+      "libpath": [],
+      "dependencies": [
+        "@stdlib/number/float32/base/to-word",
+        "@stdlib/constants/float64/high-word-abs-mask",
```
Appears to be unused.

```suggestion
```

Planeshifter committed:
fix: remove unused imports

Signed-off-by: Philipp Burckhardt <pburckhardt@outlook.com>

Planeshifter reviewed (changes requested):
One concern about `main.js` implementation that I only noticed after making one edit to the `manifest.json` and was about to approve and merge; otherwise looks great to me!

Planeshifter reviewed (dismissed):
Upon reflection, looks all fine to me!

anandkaranubc committed:
Merge branch 'develop' into feat/sinf

anandkaranubc committed:
chore: apply suggestions from review

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: passed
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

anandkaranubc commented:
@Planeshifter Thanks for the review. I also made the changes to the C benchmarks.

Planeshifter reviewed (approved):
LGTM!
